### ISSUE TITLE
RandomTeleport ignores most config options and safeteleport

### ISSUE THREAD

**Issue by spathizilla** [Zrips/CMI#3043]
**Description of issue:**
RandomTeleport ignores Y settings and also will drop into lava regardless of settings or the safeteleport permission

RT resulted in me landing at:
* 1331, 11, 519 -- WRONG
* 403, 29, 2352 -- WRONG
* 575, 63, -575 -- Correct
* 2973, 61, -2183 -- WRONG and teleported inside a block!!!
* 1292, 75, -1061 -- Teleported inside a block

**CONFIG SECTION (DELETE IF NOT RELEVANT):**
```
    world:
      Enabled: true
      MaxRange: 3000
      MinRange: 0
      CenterX: 0
      CenterZ: 0
      Circle: false
      IgnoreWater: true
      IgnoreLava: true
      minY: 63
      maxY: 80
```
cmi.safeteleport is set but /cmi checkperm safeteleport shows it as orange. I most definitely have the permission (op alone should grant it but specifically set it on myself with luckperms)

---

[16:05:07 INFO]: CMI: 8.6.21.2 BungeeCord
[16:05:07 INFO]: Server version: Paper(161) 1.15.2-R0.1-SNAPSHOT

**Comment by spathizilla** [comment]
Also tried setting:
```
      IgnoreWater: false
      IgnoreLava: false
```
without luck and still end up in lava or water. Although I'd assume it has to be true?

**Comment by spathizilla** [comment]
RandomTeleport is also not being random and keeps sending you to the same spot every so often

**Comment by HuntingYoshi** [comment]
On my server this only happens if I am in creative myself. If you are in survival you shouldn't end up inside of blocks, lava or water. You should test that

**Comment by spathizilla** [comment]
This is regular users and not me.

**Comment by Momshroom** [comment]
This started happening, as well as tping players into lava, when we updated from CMI 8.6.21.1 to 8.6.21.2.  It was not an issue before.  (though 8.6.21.1 didn't let them teleport to places that previously WERE working.). We've solved this issue by reverting to 8.6.20.9, which had the warmup issue, but at least that's fixable by letting players bypass the warmup and doesn't kill them off. :3

**Comment by Zrips** [comment]
Yea, issue confirmed and should get fixed with todays update.
